web-api/src/storage/property_filter.rs

```diff
@@ -81,6 +86,45 @@ impl IndexedPropertiesSchema {
         self.properties.extend(schema_update.clone().into_iter());
         Ok(())
     }
+
+    pub(crate) fn validate_property(
+        &self,
+        document: &DocumentId,
+        property: &DocumentPropertyId,
+        value: &DocumentProperty,
+    ) -> Result<(), InvalidDocumentProperty> {
+        // This code only checks additional validity constraints coming from on a schema
+        // but an otherwise expect a valid property, hence why we accept a `&DocumentProperty`
```

Comment:
:art: phrasing
```suggestion
        // This code only checks additional validity constraints coming from a schema
        // but otherwise expect a valid property, hence why we take a `&DocumentProperty`
```